File: python/553. Optimal Division.py
class Solution(object):
    def optimalDivision(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        n = len(nums)
        if n==1:
            return str(nums[0])
        if n==2:
            return '/'.join([str(i) for i in nums])
        return '{}/({})'.format(nums[0], '/'.join([str(i) for i in nums[1:]]))

File: go/src/leetcode/553/553. Optimal Division.go
/**
@author ZhengHao Lou
Date    2022/02/27
*/
package main

import (
    "fmt"
    "strings"
)

/**
https://leetcode-cn.com/problems/optimal-division/
思路：除数最小的时候，结果最大
*/
func optimalDivision(nums []int) string {
    n := len(nums)
    if n == 1 {
        return fmt.Sprintf("%v", nums[0])
    }
    if n == 2 {
        return fmt.Sprintf("%v/%v", nums[0], nums[1])
    }
    var strs []string
    for _, num := range nums {
        strs = append(strs, fmt.Sprintf("%v", num))
    }

    return fmt.Sprintf("%v/(%v)", strs[0], strings.Join(strs[1:], "/"))
}

func main() {
    optimalDivision([]int{1000, 100, 10, 2})
}
